store/sqlstore/file_info_store.go

```diff
@@ -217,6 +217,36 @@ func (fs SqlFileInfoStore) AttachToPost(fileId, postId string) store.StoreChanne
 	})
 }
 
+func (fs SqlFileInfoStore) AttachMultipleToPost(fileIds []string, postId string, creatorId string) store.StoreChannel {
+	return store.Do(func(result *store.StoreResult) {
+		keys, params := MapStringsToQueryParams(fileIds, "FileId")
+
+		params["PostId"] = postId
+		params["CreatorId"] = creatorId
+
+		sqlResult, err := fs.GetMaster().Exec(
+			`UPDATE
+					FileInfo
+				SET
+					PostId = :PostId
+				WHERE
+					Id IN `+keys+`
+					AND PostId = ''
+					AND CreatorId = :CreatorId`, params)
+		if err != nil {
+			result.Err = model.NewAppError("SqlFileInfoStore.AttachMultipleToPost",
+				"store.sql_file_info.attach_to_post.app_error", nil, "post_id="+postId+", err="+err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		result.Data, err = sqlResult.RowsAffected()
```

Comment:
This seems an unfortunately opaque return value: if the result actually mattered, it would be impossible to know which attachments failed. 0/5 at this point.

That being said, I'm wondering if we want to move this business logic into the App layer instead, and leave the store flexible to do whatever is needed. That would probably incur a few extra database calls (e.g. fetch `FileInfo`, then update only the right ones), but we'd also get the more explicit error handling noted above. 2/5.

And, is the `*Multiple*` strictly necessary? Was there a performance concern with looping up to 5 queries instead? 1.9999999/5.

Reply:
I tried my best have it actually return the rows added, even going so far as to spend some time bypassing gorp and messing around with Go's SQL driver, but it didn't seem possible without some extreme changes. That being said, `RowsAffected` never returns an error with MySQL and it only errors in Postgres if you try to execute the empty string as a query. I can clarify that comment to note that it only applies if we ever support additional database types.

It's not strictly necessary to have the batched method, especially since this won't even apply to most posts, but I didn't think it would hurt to switch to that since I like the convention of batching those when possible.

I'll think some more on this since the reasons against batching these changes are piling up the more I think about it.